recipes/clipper/all/conanfile.py

```diff
@@ -1,61 +1,81 @@
+    topics = ("clipper", "clipping", "polygon")
     url = "https://github.com/conan-io/conan-center-index"
     homepage = "https://github.com/skyrpex/clipper"
     license = "BSL-1.0"
-    exports_sources = ["CMakeLists.txt", "patches/*"]
-    generators = "cmake"
-    settings = "arch", "build_type", "compiler", "os"
-    _cmake = None
-    _source_subfolder = "source_subfolder"
-    _build_subfolder = "build_subfolder"
+
+    settings = "os", "arch", "compiler", "build_type"
     options = {
         "shared": [True, False],
-        "fPIC": [True, False]
+        "fPIC": [True, False],
+    }
+    default_options = {
+        "shared": False,
+        "fPIC": True,
     }
-    default_options = {"shared": False, "fPIC": True}
 
-    def source(self):
-        tools.get(**self.conan_data["sources"][self.version])
-        extracted_dir = self.name + "-" + self.version
-        os.rename(extracted_dir, self._source_subfolder)
+    generators = "cmake"
+    _cmake = None
+
+    @property
+    def _source_subfolder(self):
+        return "source_subfolder"
+
+    @property
+    def _build_subfolder(self):
+        return "build_subfolder"
+
+    def export_sources(self):
+        self.copy("CMakeLists.txt")
+        for patch in self.conan_data.get("patches", {}).get(self.version, []):
+            self.copy(patch["patch_file"])
 
     def config_options(self):
         if self.settings.os == "Windows":
             del self.options.fPIC
 
+    def configure(self):
+        if self.options.shared:
+            del self.options.fPIC
+
+    def source(self):
+        tools.get(**self.conan_data["sources"][self.version],
+                  destination=self._source_subfolder, strip_root=True)
+
     def _configure_cmake(self):
         if self._cmake:
             return self._cmake
-
         self._cmake = CMake(self)
         self._cmake.configure(build_folder=self._build_subfolder)
         return self._cmake
 
     def build(self):
-        if "patches" in self.conan_data and self.version in self.conan_data["patches"]:
-            for patch in self.conan_data["patches"][self.version]:
-                tools.patch(**patch)
-
+        for patch in self.conan_data.get("patches", {}).get(self.version, []):
+            tools.patch(**patch)
         cmake = self._configure_cmake()
         cmake.build()
 
     def package(self):
         cmake = self._configure_cmake()
         cmake.install()
-
         self.copy("License.txt", dst="licenses", src=self._source_subfolder)
         tools.rmdir(os.path.join(self.package_folder, "share"))
-        tools.rmdir(os.path.join(self.package_folder, "lib", "cmake"))
 
     def package_info(self):
+        self.cpp_info.set_property("pkg_config_name", "polyclipping")
+        self.cpp_info.libs = ["polyclipping"]
+
+        # TODO: to remove in conan v2 once cmake_find_package* & pkg_config generators removed.
+        #       Do not use these CMake names in CMakeDeps, it was a mistake,
```

Comment:
Thanks for your amazing effort on these targets 

```suggestion
        # But keep the rest so we dont make this mistake again
        #       Do not use these CMake names in CMakeDeps, it was a mistake,
```